mantidproject/mantid
# Xray absorption correction Algorithm

**Description of work.**

Adds a new algorithm called XrayAbsorptionCorrection to be used for muon elemental analysis.

**To test:**

1. Add Sundials to in Data search directory(Contact myself or @AnthonyLim23).
2. Run script below to load data, create muon implantation profile , set sample and sample material and run XrayAbsorptionCorrection algorithm, and should run to produce a workspace called correction containing the correction factors.
```
LoadElementalAnalysisData(Run='2695', GroupWorkspace='ws')
CreateWorkspace(OutputWorkspace='MuonProfile', DataY = np.arange(1,51), DataX = np.linspace(0.1,0.2))
SetSample(InputWorkspace='ws; Detector 1', Geometry='{"Center":[0.0,0.0,0.0],"Height":2.0,"Radius":1.0,"Shape":"Cylinder"}')
SetSampleMaterial(InputWorkspace='ws; Detector 1', ChemicalFormula='Au', 
XRayAttenuationProfile=path to absorption coefficient)
XrayAbsorptionCorrection(InputWorkspace='ws; Detector1',
MuonImplantationProfile='MuonProfile',OutputWorkspace='corrections')
```
3. Run script without setsample algorithm and setsample material algorithm, and it should produce an error message.

4. Run script without setsamplematerial algorithm, and it should produce an error message.

There is no associated issue.





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
